lib/Sema/TypeCheckType.cpp

```diff
@@ -1382,6 +1382,19 @@ static Type diagnoseUnknownType(TypeResolution resolution,
 
       return I->second;
     }
+    
+    // type-casting operators such as 'is' and 'as'.
+    if (resolution.getOptions().is(TypeResolverContext::ExplicitCastExpr)) {
+      auto lookupResult = TypeChecker::lookupUnqualified(
+          dc, repr->getNameRef(), repr->getLoc(), lookupOptions);
+      if (!lookupResult.empty()) {
+        auto first = lookupResult.front().getValueDecl();
+        diags.diagnose(L, diag::cannot_find_type_in_cast_expression, first)
+          .highlight(R);
+        diags.diagnose(first->getNameLoc(), diag::note_declared_here);
```

Comment:
Two notices here:

- Let's just use `diags.diagnose(first, ...)`. We want to point to the declaration site.
- There is a "declared here" note already - https://github.com/apple/swift/blob/main/include/swift/AST/DiagnosticsSema.def#L29-L30

Reply:
Done.